app/pages/learn/01_tutorial/01_your-first-java-app/02_building-with-intellij-idea.md

```diff
@@ -0,0 +1,262 @@
+
+Alternatively, we can run our application using the green Run button in the top right corner, or using the shortcut **⌃R** (on macOS) or **Ctrl+F10** (on Windows/Linux) to run the latest file.
+
+[![Run](/assets/images/intellij-idea/run.png)](/assets/images/intellij-idea/run.png)
+
+If we want to pass arguments to our application, we can do so in our [Run Configurations](https://www.jetbrains.com/help/idea/run-debug-configuration.html).
+
+To edit your run configurations, select the configuration in the run/debug configuration switcher, by clicking the down arrow next to the current configuration or the three dots to the right of the run configuration, and select **Edit Configurations**.
+
+[![Edit Configurations](/assets/images/intellij-idea/edit-configurations.png)](/assets/images/intellij-idea/edit-configurations.png)
+
+Edit the **Run/Debug Configurations** in the popup.
+
+[![Run / Debug Configuration](/assets/images/intellij-idea/run-config.png)](/assets/images/intellij-idea/run-config.png)
+
+## Testing
+Testing your code helps you to verify that the code does what you expect it to do. You can run your application and test it yourself, or add automated tests that can verify your code for you. Thinking about what to test and how, can help you to break a problem up into smaller pieces. This will help you get a better solution faster!
+
+For example, let's say we have the following method that calculates the average of a list of values, and we want to make sure the average is calculated correctly:
+```java
+public class Average {
+
+    public static double calculateAverage(int[] numbers) {
+        int sum = 0;
+        for (int number : numbers) {
+            sum += number;
+        }
+        double avg = (double) sum / numbers.length;
+        return avg;
+    }
+}
+```
+IntelliJ IDEA makes it easy to add tests to your code. You can navigate to the test for a particular class using the shortcut **⇧⌘T** on macOS or **Ctrl+Shift+T** on Windows/Linux. If no test class exists yet, IntelliJ IDEA will create one for you. This class will be created in the `src/main/java` directory.
+We can select a **Testing library** in the **Create test** popup.
+
+[![Create test](/assets/images/intellij-idea/create-test.png)](/assets/images/intellij-idea/create-test.png)
+
+IntelliJ IDEA supports multiple testing libraries, including [JUnit 5](https://junit.org/JUnit5/), which is the [most used testing library for Java developers](https://www.jetbrains.com/lp/devecosystem-2023/java/#java_unittesting). If JUnit 5 is not part of your project yet, IntelliJ IDEA will note “JUnit5 library not found in the module”. Click **Fix** to have IntelliJ IDEA fix this for you.
+
+Note that the JUnit 5 dependency `junit-jupiter` is added to the `pom.xml` in the `<dependencies>` section.
+
+[![JUnit5 dependencies](/assets/images/intellij-idea/JUnit5-dependencies.png)](/assets/images/intellij-idea/JUnit5-dependencies.png)
```

Comment:
This image doesn't work.
![image](https://github.com/java/devjava-content/assets/34687786/1d575b38-e9e3-4ad9-a310-6955c6d869cf)

Reply:
Odd, as it works for me
<img width="987" alt="Screenshot 2024-04-05 at 13 55 37" src="https://github.com/java/devjava-content/assets/20927578/8155e9d7-5b93-4d2f-8384-1def43dd9120">

Reply:
I think this might be due to case-sensitivity. The file is named `junit5-dependencies.png` while the link refers to `JUnit5-dependencies.png`.

Reply:
Fixed file name, please try again?